src/dfx/src/commands/canister/request_status.rs

```diff
@@ -46,46 +48,61 @@ pub async fn exec(env: &dyn Environment, opts: RequestStatusOpts) -> DfxResult {
     let canister_id = Principal::from_text(callee_canister)
         .or_else(|_| canister_id_store.get(callee_canister))?;
 
-    let mut retry_policy = ExponentialBackoff::default();
-    let blob = async {
-        let mut request_accepted = false;
-        loop {
-            let (response, _cert) = agent
-                .request_status_raw(&request_id, canister_id)
-                .await
-                .context("Failed to fetch request status.")?;
-            match response {
-                RequestStatusResponse::Replied(reply) => return Ok(reply.arg),
-                RequestStatusResponse::Rejected(response) => {
-                    return Err(DfxError::new(AgentError::CertifiedReject(response)))
-                }
-                RequestStatusResponse::Unknown => (),
-                RequestStatusResponse::Received | RequestStatusResponse::Processing => {
-                    // The system will return Unknown until the request is accepted
-                    // and we generally cannot know how long that will take.
-                    // State transitions between Received and Processing may be
-                    // instantaneous. Therefore, once we know the request is accepted,
-                    // we restart the waiter so the request does not time out.
-                    if !request_accepted {
-                        retry_policy.reset();
-                        request_accepted = true;
+    let blob = if let Some(pocketic) = env.get_pocketic() {
+        let msg_id = RawMessageId {
+            effective_principal: RawEffectivePrincipal::CanisterId(canister_id.as_slice().to_vec()),
+            message_id: request_id.as_slice().to_vec(),
+        };
+        let res = pocketic
+            .await_call_no_ticks(msg_id)
+            .await
+            .map_err(|err| anyhow!("Canister call failed: {}", err))?;
```

Comment:
```suggestion
            .map_err(|err| anyhow!("Canister request status failed: {}", err))?;
```

Reply:
Due to how PocketIC call response types are defined at the moment (see [forum](https://forum.dfinity.org/t/pocketic-call-response-types/38728)), canister call errors are represented by `Err(...)` - this line, e.g., canister trapped - as well as `Ok(WasmResult::Reject(...))` - below, e.g., canister call `ic_cdk::api::call::reject`.